app/controllers/newsletters_controller.rb

```diff
@@ -0,0 +1,6 @@
+class NewslettersController < ApplicationController
+  def index
+    @flyers = NewsletterFinder.new.flyers
```

Comment:
I have heard of this principle. As it happens I have just started the chapter in POODR where she explains it. I will continue reading and try to apply it. 
Hopefully I can come back to you with something cleaner.